arangod/Transaction/Methods.cpp

```diff
@@ -1510,28 +1514,39 @@ Future<OperationResult> transaction::Methods::modifyLocal(
                             options.returnNew ? &result : nullptr);
     }
 
+    if (result.revisionId() == previous.revisionId()) {
+      excludeFromReplication = true;
+    }
+
     return res;  // must be ok!
   };             // workForOneDocument
   ///////////////////////
 
   Result res;
   std::unordered_map<ErrorCode, size_t> errorCounter;
 
+  std::unordered_set<size_t> excludePositions;
   if (newValue.isArray()) {
     VPackArrayBuilder guard(&resultBuilder);
     VPackArrayIterator it(newValue);
     while (it.valid()) {
-      res = workForOneDocument(it.value(), true);
+      bool excludeFromReplication = false;
+      res = workForOneDocument(it.value(), true, excludeFromReplication);
       if (res.fail()) {
         createBabiesError(resultBuilder, errorCounter, res);
+      } else if (excludeFromReplication) {
+        excludePositions.insert(it.index());
       }
       it.next();
     }
-
     // it is ok to clobber res here!
     res.reset();
   } else {
-    res = workForOneDocument(newValue, false);
+    bool excludeFromReplication = false;
+    res = workForOneDocument(newValue, false, excludeFromReplication);
+    if (res.ok() && excludeFromReplication) {
+      excludePositions.insert(0);
```

Comment:
Ok, `excludePositions` are populated here, but later on nothing happens with them.
Especially, **they are not passed to replicateOperations**, so they are effectively useless.